pkg/helm/fake.go

```diff
@@ -135,14 +156,44 @@ func (c *FakeClient) UpdateRelease(rlsName string, chStr string, opts ...UpdateO
 }
 
 // UpdateReleaseFromChart returns an UpdateReleaseResponse containing the updated release, if it exists
-func (c *FakeClient) UpdateReleaseFromChart(rlsName string, chart *chart.Chart, opts ...UpdateOption) (*rls.UpdateReleaseResponse, error) {
+func (c *FakeClient) UpdateReleaseFromChart(rlsName string, newChart *chart.Chart, opts ...UpdateOption) (*rls.UpdateReleaseResponse, error) {
+	for _, opt := range opts {
+		opt(&c.Opts)
+	}
 	// Check to see if the release already exists.
 	rel, err := c.ReleaseContent(rlsName, nil)
 	if err != nil {
 		return nil, err
 	}
 
-	return &rls.UpdateReleaseResponse{Release: rel.Release}, nil
+	mockOpts := &MockReleaseOptions{
+		Name:        rel.Release.Name,
+		Version:     rel.Release.Version + 1,
+		Chart:       newChart,
+		Config:      c.Opts.updateReq.Values,
+		Namespace:   rel.Release.Namespace,
+		Description: c.Opts.updateReq.Description,
+	}
+
+	newRelease := ReleaseMock(mockOpts)
+
+	if c.Opts.updateReq.ResetValues {
+		newRelease.Config = &chart.Config{Raw: "{}"}
+	} else if c.Opts.updateReq.ReuseValues {
+		// TODO: This should merge old and new values but does not.
```

Comment:
I left it unimplemented in this PR intentionally only because the `reuseValues` logic in the tiller is kind of complex, and I think the real fix is to export that function from the tiller and just use it, since it's not really tied to the tiller / releaseServer at all, except to use the logger. Was going to save that fight till the next one :D

Reply:
Ok, that sounds fair. 😄